### ISSUE TITLE
Collateral goes missing when submitting to Ogmios

### ISSUE THREAD

**Issue by vvtran** [Plutonomicon/cardano-browser-tx#259]
On branch `viet/seabug-address-fixes`. Running `npm run dev` and signing builds, balances and signs a transaction. Closer inspection of transaction suggests it is indeed balancing correctly. Here is the entire `Transaction` log
```
(Transaction { auxiliary_data: Nothing, body: (TxBody { auxiliary_data_hash: Nothing, certs: Nothing, collateral: (Just [(TransactionInput { index: 0u, transaction_id: (TransactionHash (byteArrayFromIntArrayUnsafe [180,118,245,6,211,153,203,40,137,14,84,193,134,206,114,200,197,202,79,101,208,187,247,156,184,125,64,75,98,49,68,156])) })]), fee: (Coin fromString "224345"), inputs: [(TransactionInput { index: 0u, transaction_id: (TransactionHash (byteArrayFromIntArrayUnsafe [136,232,42,237,69,157,75,217,133,105,126,79,136,20,50,30,14,42,134,4,102,42,232,167,136,170,85,213,122,153,140,1])) }),(TransactionInput { index: 1u, transaction_id: (TransactionHash (byteArrayFromIntArrayUnsafe [25,239,161,222,238,219,30,111,72,190,245,138,170,170,65,197,120,116,169,65,206,255,108,69,224,120,154,24,181,223,156,26])) })], mint: (Just (Mint (Value (Coin fromString "0") (NonAdaAsset(fromFoldable [(Tuple (CurrencySymbol(byteArrayFromIntArrayUnsafe [112,64,99,103,48,231,58,234,5,76,11,45,208,183,52,190,195,236,170,202,30,60,190,72,180,130,202,20])) (fromFoldable [(Tuple (TokenName(byteArrayFromIntArrayUnsafe [65,162,33,93,139,129,115,44,40,195,210,64,108,84,6,20,181,35,64,20,213,93,161,7,113,45,189,216,12,228,122,146])) fromString "1"),(Tuple (TokenName(byteArrayFromIntArrayUnsafe [216,80,215,167,15,213,255,151,171,16,75,18,125,76,150,48,198,77,138,193,88,161,76,220,196,246,81,87,183,154,240,186])) fromString "-1")]))]))))), network_id: (Just TestnetId), outputs: [(TransactionOutput { address: (Address addr_test1qpkl55y6av6lvu9gfjkqr0cmtkxn7qcezgg0q3vr0m2huda6qh8x3elu6qa9t48ymn0dluh805ws2we38uxwxcnzyygq5jl946), amount: (Value (Coin fromString "1890901712") (NonAdaAsset(fromFoldable [(Tuple (CurrencySymbol(byteArrayFromIntArrayUnsafe [112,64,99,103,48,231,58,234,5,76,11,45,208,183,52,190,195,236,170,202,30,60,190,72,180,130,202,20])) (fromFoldable [(Tuple (TokenName(byteArrayFromIntArrayUnsafe [216,80,215,167,15,213,255,151,171,16,75,18,125,76,150,48,198,77,138,193,88,161,76,220,196,246,81,87,183,154,240,186])) fromString "0")]))]))), data_hash: Nothing }),(TransactionOutput { address: (Address addr_test1wr05mmuhd3nvyjan9u4a7c76gj756am40qg7vuz90vnkjzczfulda), amount: (Value (Coin fromString "2000000") (NonAdaAsset(fromFoldable [(Tuple (CurrencySymbol(byteArrayFromIntArrayUnsafe [112,64,99,103,48,231,58,234,5,76,11,45,208,183,52,190,195,236,170,202,30,60,190,72,180,130,202,20])) (fromFoldable [(Tuple (TokenName(byteArrayFromIntArrayUnsafe [65,162,33,93,139,129,115,44,40,195,210,64,108,84,6,20,181,35,64,20,213,93,161,7,113,45,189,216,12,228,122,146])) fromString "1")]))]))), data_hash: (Just (DataHash (byteArrayFromIntArrayUnsafe [88,32,228,248,189,201,178,178,171,114,109,201,150,232,70,140,87,138,105,243,65,145,191,9,165,110,106,28,102,29,38,148,224,223]))) }),(TransactionOutput { address: (Address addr_test1vqlnger9p044xfxsu337h6qlhc0a2xdkgwzjr6tdp56m6agyumjux), amount: (Value (Coin fromString "85000000") (NonAdaAsset(fromFoldable []))), data_hash: (Just (DataHash (byteArrayFromIntArrayUnsafe [88,32,150,29,126,189,56,59,239,17,110,249,188,181,39,241,188,26,69,135,248,25,212,236,14,114,63,239,90,178,40,173,29,143]))) }),(TransactionOutput { address: (Address addr_test1vqlnger9p044xfxsu337h6qlhc0a2xdkgwzjr6tdp56m6agyumjux), amount: (Value (Coin fromString "10000000") (NonAdaAsset(fromFoldable []))), data_hash: (Just (DataHash (byteArrayFromIntArrayUnsafe [88,32,150,29,126,189,56,59,239,17,110,249,188,181,39,241,188,26,69,135,248,25,212,236,14,114,63,239,90,178,40,173,29,143]))) }),(TransactionOutput { address: (Address addr_test1wzw637nk52s02249mug0k7uplxh7fvswjp5t872l4hr5w7se4c6r0), amount: (Value (Coin fromString "5000000") (NonAdaAsset(fromFoldable []))), data_hash: (Just (DataHash (byteArrayFromIntArrayUnsafe [88,32,150,29,126,189,56,59,239,17,110,249,188,181,39,241,188,26,69,135,248,25,212,236,14,114,63,239,90,178,40,173,29,143]))) })], required_signers: Nothing, script_data_hash: Nothing, ttl: Nothing, update: Nothing, validity_start_interval: Nothing, withdrawals: Nothing }), is_valid: true, witness_set: (TransactionWitnessSet { bootstraps: Nothing, native_scripts: Nothing, plutus_data: Nothing, plutus_scripts: Nothing, redeemers: Nothing, vkeys: (Just [(Vkeywitness (Tuple (Vkey (PublicKey "ed25519_pk13hef55hmu0w39663gtf3yc56yyy7ln4ztcjhgndclzr39rcu0nmqw0swny")) (Ed25519Signature "ed25519_sig17nj6cskdzm7harxw33wcl4qxl5lmqp9y0n8h6gy2fmayg0nqy4hkd0vzhj7nmudjjyj3uyyeyygltrqhhz6uv3wlpcv8kq3yuqa3yzgc82rve")))]) }) })
```
where
`addr_test1wr05mmuhd3nvyjan9u4a7c76gj756am40qg7vuz90vnkjzczfulda` = validator address
`addr_test1vqlnger9p044xfxsu337h6qlhc0a2xdkgwzjr6tdp56m6agyumjux` = author address = owner address
`addr_test1wzw637nk52s02249mug0k7uplxh7fvswjp5t872l4hr5w7se4c6r0` = dao script address
`addr_test1qpkl55y6av6lvu9gfjkqr0cmtkxn7qcezgg0q3vr0m2huda6qh8x3elu6qa9t48ymn0dluh805ws2we38uxwxcnzyygq5jl946` = me

As far as I can tell, everything is in place, the `collateral` is there. However we get an error from Ogmios
```
unexpected parse error on input:{"type":"jsonwsp/response","version":"1.0","servicename":"ogmios","methodname":"SubmitTx","result":{"SubmitFail":[{"unspendableDatums":{"nonSpendable":["961d7ebd383bef116ef9bcb527f1bc1a4587f819d4ec0e723fef5ab228ad1d8f","e4f8bdc9b2b2ab726dc996e8468c578a69f34191bf09a56e6a1c661d2694e0df"],"acceptable":[]}},{"extraRedeemers":["spend:0","mint:0"]},{"extraDataMismatch":{"provided":"2b3ae1354dd35f7b506335ffa158f996e1ec9db0d717354d771b02454cfac249","inferredFromParameters":"e502d22f3800ff6101b5a8896c55e8d90101a5ae3902014b2c86b863bbfb92c8"}},{"missingScriptWitnesses":["df4def976c66c24bb32f2bdf63da44bd4d77757811e670457b27690b"]},{"invalidWitnesses":["8df29a52fbe3dd12eb5142d312629a2109efcea25e25744db8f887128f1c7cf6"]},{"collectErrors":[{"noWitness":"df4def976c66c24bb32f2bdf63da44bd4d77757811e670457b27690b"}]},{"valueNotConserved":{"consumed":{"coins":1993126057,"assets":{"7040636730e73aea054c0b2dd0b734bec3ecaaca1e3cbe48b482ca14.d850d7a70fd5ff97ab104b127d4c9630c64d8ac158a14cdcc4f65157b79af0ba":1}},"produced":{"coins":1993126057,"assets":{"7040636730e73aea054c0b2dd0b734bec3ecaaca1e3cbe48b482ca14.41a2215d8b81732c28c3d2406c540614b5234014d55da107712dbdd80ce47a92":1}}}},{"missingCollateralInputs":null},{"collateralTooSmall":{"requiredCollateral":336518,"actualCollateral":0}},{"feeTooSmall":{"requiredFee":568221,"actualFee":224345}}]},"reflection":{"id":"SubmitTx-l1gicrni","step":"INIT"}}
```
It's saying the fee is too small, but I think this is a result of `"collateralTooSmall":{"requiredCollateral":336518,"actualCollateral":0}}` not having any collateral, which contradicts what happens after balancing. `336518 + 224345 = 560863` which is fairly closed to the `requiredFee` `568221`.

**Comment by vvtran** [comment]
Some things to investigate on Monday before I forget:
- Check min Ada requirements (logic and tx state after balancing).
- Another close look at the balanced tx body & compare to constraints.
- Check front end server call (Affjax).
- Check backend server (cbor decoding, attaching then encoding).

**Comment by ngua** [comment]
Fixed by #260
